openc3/python/openc3/utilities/connection_pool.py

```diff
@@ -37,5 +38,7 @@ def get(self):
                 self.count += 1
             else:
                 item = self.pool.get()
-        yield item
-        self.pool.put(item)
+        try:
+            yield item
+        finally:
+            self.pool.put(item)
```

Comment:
Was this losing items if the yield raised an exception?

Reply:
I don't know where it was losing items, but it definitely was.  This was eventually causing the decom lockup because it was waiting forever for a redis instance to be available.